src/main/java/Duke.java

```diff
@@ -1,10 +1,249 @@
     public static void main(String[] args) {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("---------------------------------------------------------------------------------");
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        System.out.println("---------------------------------------------------------------------------------");
+        userInput = in.nextLine();
+        while (true) { // ensure that the loop can stay on forever if needed.
+            while(userInput.equals("") || userInput.equals(" ")) {
+                printLine();
+                System.out.println("\tSorry please enter a valid input ");
+                printLine();
+                userInput = in.nextLine();
+            }
+            if(userInput.equals("bye")) { // exit command
+                break;
+            } else if(userInput.equals("list")) { //displays the list if needed
+                printLine();
+                System.out.println("\tHere are the tasks in your list:");
+                listTasks(currentIndex, taskList);
+                printLine();
+            } else if (isTheSame(userInput, "mark")) { //mark the task in
+                while(isInRange(userInput, currentIndex)==false) {
+                    printLine();
+                    System.out.println("\tNice try, enter a valid index to mark:");
+                    printLine();
+                    userInput = in.nextLine();
+                }
+                printLine();
+                printMarkedTask(userInput, taskList);
+                printLine();
+            } else if (isTheSame(userInput, "unmark")) {//unmark the task
+                while(isInRange(userInput, currentIndex)==false) {
+                    printLine();
+                    System.out.println("\tNice try, enter a valid index to unmark:");
+                    printLine();
+                    userInput = in.nextLine();
+                }
+                printLine();
+                printUnmarkedTask(userInput, taskList);
+                printLine();
+            } else if(isTheSame(userInput, "todo")) {
+                todoExceptionHandler();
+                //leave this for the final refactoring
+            } else if(isTheSame(userInput, "deadline")) {
+                deadlineExceptionHandler();
+            } else if(isTheSame(userInput, "event")) {
+                eventExceptionHandler();
+            }
+            else { // tells the user that we have added the task in
+                printTask(userInput); // could remove this and esnure that only specific tasks can be entered!
+            }
+            userInput = in.nextLine();
+        }
+       sayBye();
```

Comment:
Methods should not be greater than 30 LOC. You can split the code here into multiple methods to improve readability.